apps/api/src/app/events/usecases/send-message/send-message.usecase.ts

```diff
@@ -35,4 +44,19 @@ export class SendMessage {
         return await this.digest.execute(command);
     }
   }
+
+  private async filter(command: SendMessageCommand) {
+    const data = await this.getFilterData(command);
+
+    return matchMessageWithFilters(command.step, data);
+  }
+
+  private async getFilterData(command: SendMessageCommand) {
+    const subscriber = await this.subscriberRepository.findById(command.subscriberId);
```

Comment:
Is the findById looks by subscriberId? Or is it _subscriberId. Also is this query is mandatory always or only when we have the `subscriber` filters? Can we only fetch it when needed?